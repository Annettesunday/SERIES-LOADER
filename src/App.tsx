import React from "react";
import { Link } from "@reach/router";
import { Store } from "./Store";

export default function App(props: any): JSX.Element {
  const { state } = React.useContext(Store);
  return (
    <React.Fragment>
      <header className="header">
        <div>
          <h1 data-testid="header-test">Rick and Morty</h1>
          <p>Pick your favorite episode</p>
        </div>
        <div>
          <Link to="/">Home</Link>{" "}
          <Link to="/faves" data-testid="fav-test">
            Favorite(s): {state && state.favorites.length}
          </Link>
        </div>
      </header>
      {props.children}
    </React.Fragment>
  );
}
