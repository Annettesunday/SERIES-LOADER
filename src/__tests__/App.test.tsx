import React from "react";
import { render, cleanup } from "@testing-library/react";
import App from "../App";
import { async } from "q";

afterEach(cleanup);

describe("<App />", () => {
  function renderAppComponent() {
    return render(<App />);
  }
  test("should match snapshot", () => {
    const { asFragment } = renderAppComponent();
    expect(asFragment()).toMatchSnapshot();
  });
  test("favorites should be set to 0 when App first renders", async () => {
    const { findByTestId } = renderAppComponent();
    const favoritesValue = await findByTestId("fav-test");
    // assert that it must be 0
    expect(favoritesValue.innerHTML).toBe("Favorite(s): ");
  });
  test("that header is loaded when the App renders", async () => {
    const { findByTestId } = renderAppComponent();
    const header = await findByTestId("header-test");
    expect(header.innerHTML).toBe("Rick and Morty");
  });
});
