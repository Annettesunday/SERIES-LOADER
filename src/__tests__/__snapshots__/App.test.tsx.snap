// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <div>
      <h1
        data-testid="header-test"
      >
        Rick and Morty
      </h1>
      <p>
        Pick your favorite episode
      </p>
    </div>
    <div>
      <a
        aria-current="page"
        href="/"
      >
        Home
      </a>
       
      <a
        data-testid="fav-test"
        href="/faves"
      >
        Favorite(s): 
      </a>
    </div>
  </header>
</DocumentFragment>
`;
